Anonymous class in higher kinded value class

## Compiler version

3.3.2-RC1-bin-20230628-f3787eb-NIGHTLY and before

## Minimized code

```scala
class Foo[A]

object Test {
  class Bar[F[_]](val underlying: Any) extends AnyVal {
    def foo = new Foo[F[Any]] {}
  }
}
```

## Output (click arrow to expand)
<details>

```scala
java.lang.AssertionError: assertion failed: Unexpected alias type: type F
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at dotty.tools.dotc.transform.GenericSignatures$.jsig$1(GenericSignatures.scala:261)
        at dotty.tools.dotc.transform.GenericSignatures$.boxedSig$1(GenericSignatures.scala:68)
        at dotty.tools.dotc.transform.GenericSignatures$.argSig$1(GenericSignatures.scala:195)
        at dotty.tools.dotc.transform.GenericSignatures$.classSig$1$$anonfun$1(GenericSignatures.scala:229)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at dotty.tools.dotc.transform.GenericSignatures$.classSig$1(GenericSignatures.scala:229)
        at dotty.tools.dotc.transform.GenericSignatures$.jsig$1(GenericSignatures.scala:288)
        at dotty.tools.dotc.transform.GenericSignatures$.boxedSig$1(GenericSignatures.scala:68)
        at dotty.tools.dotc.transform.GenericSignatures$.superSig$1$$anonfun$1(GenericSignatures.scala:65)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at dotty.tools.dotc.transform.GenericSignatures$.superSig$1(GenericSignatures.scala:65)
        at dotty.tools.dotc.transform.GenericSignatures$.jsig$1(GenericSignatures.scala:348)
        at dotty.tools.dotc.transform.GenericSignatures$.javaSig0(GenericSignatures.scala:365)
        at dotty.tools.dotc.transform.GenericSignatures$.javaSig(GenericSignatures.scala:40)
        at dotty.tools.backend.jvm.BCodeHelpers.dotty$tools$backend$jvm$BCodeHelpers$$getGenericSignatureHelper(BCodeHelpers.scala:773)
        at dotty.tools.backend.jvm.BCodeHelpers$BCJGenSigGen.getGenericSignature(BCodeHelpers.scala:400)
        at dotty.tools.backend.jvm.BCodeHelpers$BCJGenSigGen.getGenericSignature$(BCodeHelpers.scala:381)
        at dotty.tools.backend.jvm.BCodeSkelBuilder$PlainSkelBuilder.getGenericSignature(BCodeSkelBuilder.scala:74)
        at dotty.tools.backend.jvm.BCodeSkelBuilder$PlainSkelBuilder.initJClass(BCodeSkelBuilder.scala:273)
        at dotty.tools.backend.jvm.BCodeSkelBuilder$PlainSkelBuilder.genPlainClass(BCodeSkelBuilder.scala:123)
        at dotty.tools.backend.jvm.CodeGen.genClass(CodeGen.scala:138)
        at dotty.tools.backend.jvm.CodeGen.genClassDef$1(CodeGen.scala:55)
        at dotty.tools.backend.jvm.CodeGen.genClassDefs$1(CodeGen.scala:101)
        at dotty.tools.backend.jvm.CodeGen.genClassDefs$1$$anonfun$1(CodeGen.scala:99)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at dotty.tools.backend.jvm.CodeGen.genClassDefs$1(CodeGen.scala:99)
        at dotty.tools.backend.jvm.CodeGen.genUnit(CodeGen.scala:104)
        at dotty.tools.backend.jvm.GenBCode.run(GenBCode.scala:76)
        at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:327)
        at scala.collection.immutable.List.map(List.scala:246)
        at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:331)
        at dotty.tools.backend.jvm.GenBCode.runOn(GenBCode.scala:87)
        at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:246)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
        at dotty.tools.dotc.Run.runPhases$1(Run.scala:262)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:270)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:279)
        at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
        at dotty.tools.dotc.Run.compileUnits(Run.scala:279)
        at dotty.tools.dotc.Run.compileSources(Run.scala:194)
        at dotty.tools.dotc.Run.compile(Run.scala:179)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
        at dotty.tools.dotc.Driver.process(Driver.scala:197)
        at dotty.tools.dotc.Driver.process(Driver.scala:165)
        at dotty.tools.dotc.Driver.process(Driver.scala:177)
        at dotty.tools.dotc.Driver.main(Driver.scala:207)
        at dotty.tools.dotc.Main.main(Main.scala)
Error while emitting HigherKindedValueClass.scala
assertion failed: Unexpected alias type: type F
1 error found
Compilation failed
```
</details>

It seems like the reference to `F` doesn't get erased properly, breaking bytecode generation. The code compiles successfully when `extends AnyVal` or `{}` get commented out.

--- a/tests/pos/i18099.scala
+++ b/tests/pos/i18099.scala
@@ -1,0 +1,7 @@
+class Foo[A]
+
+object Test {
+  class Bar[F[_]](val underlying: Any) extends AnyVal {
+    def foo = new Foo[F[Any]] {}
+  }
+}

